enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1078 @@
+    // control plane and compute nodes.
+    ClusterNodeSubnetRole SubnetRole = "ClusterNode"
+
+    // IngressControllerLBSubnetRole specifies subnets used by the default IngressController.
+    IngressControllerLBSubnetRole SubnetRole = "IngressControllerLB"
+
+    // ControlPlaneExternalLBSubnetRole specifies subnets used by the external control plane
+    // load balancer that serves the Kubernetes API server.
+    ControlPlaneExternalLBSubnetRole SubnetRole = "ControlPlaneExternalLB" 
+
+    // ControlPlaneInternalLBSubnetRole specifies subnets used by the internal control plane
+    // load balancer that serves the Kubernetes API server.
+    ControlPlaneInternalLBSubnetRole SubnetRole = "ControlPlaneInternalLB"
+)
+```
+
+The `platform.aws.vpc.subnets` field will be introduced under the TechPreview feature gate, and `platform.aws.subnets`
+will be deprecated when `platform.aws.vpc.subnets` graduates to GA.
+
+##### Installer Validation Rules
+
+These validation rules only apply to the new `platform.aws.vpc.subnets` field. These validations aim to improve
+user experience, but are not permanent and may be updated or removed to accommodate future use cases.
+
+###### All or Nothing Subnet Roles Selection
+
+This proposed API follows an "all or nothing" approach for roles: either all roles must be explicitly assigned
+(manual role selection), or no roles must be assigned (automatic role selection). The installer must not allow a mix of
+automatic and manual role assignments, as this could lead to confusing behavior. See [Allowing Mixed Automatic and Manual Subnet Selection Alternative](#allowing-mixed-automatic-and-manual-subnet-selection-alternative) for the reason behind this validation.
+
+However, one exception is if the cluster is private (`installconfig.publish: Internal`), then
+ControlPlaneExternalLB can be omitted.
+
+###### All Subnets Belong to the Same VPC Validation
+
+The [existing validation](https://github.com/openshift/installer/blob/0d77aa8df5ddc68e926aa11da24a87981021b256/pkg/asset/installconfig/aws/subnet.go#L91)
+for the deprecated `platform.aws.subnets` field ensuring all subnets belong to the same VPC
+should apply to all subnets specified in `platform.aws.vpc.subnets`.
+
+###### Multiple Load Balancer Subnets in the Same AZ Validation
+
+The installer must reject multiple IngressControllerLB subnets in the same AZ as this will be
+rejected by the AWS CCM.
+
+###### The Old and New Subnets Fields Cannot be Specified Together Validation
+
+Since `platform.aws.subnets` will be deprecated, the installer must not allow `platform.aws.vpc.subnets` and
+`platform.aws.subnets` to both be specified.
+
+###### Maximum of 10 IngressController Subnets Validation
+
+Since the IngressController's API [only allows 10 subnets](https://github.com/openshift/api/blob/ee5bb7eaf6b6638d4e3b33ba4ff0834212cdb75d/operator/v1/types_ingress.go#L564),
+the installer must not allow more than 10 IngressController subnets as well.
```

Comment:
Is this also the max count for subnet with other roles?

Reply:
No, just a restriction on the IngressControllerLB subnets.

Control Plane Load Balanacers subnets don't have a limit restriction enforced by the [CAPA API](https://github.com/openshift/installer/blob/6f39575f584a3fe5304f0cb08baa982026bb3bb9/vendor/sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2/awscluster_types.go#L215). And I don't see any subnet count limit for compute or control plane nodes in the Installer.

But either way, you won't be able to specify duplicate AZs (see existing [validateDuplicateSubnetZones](https://github.com/openshift/installer/blob/358ca0f104e80ce081f954b70d81a10a23617971/pkg/asset/installconfig/aws/validation.go#L451) for each role. I'll add a small snippet about this in the EP.

Reply:
Prefect. Thanks!